pkg/trace/agent/agent.go

```diff
@@ -112,6 +112,21 @@ func (a *Agent) Run() {
 	a.loop()
 }
 
+// FlushSync flushes traces sychronously. This method only works when the agent is configured in synchronous flushing
+// mode.
+func (a *Agent) FlushSync() {
```

Comment:
Let's just return immediately if it's disabled.
```suggestion
func (a *Agent) FlushSync() {
    if a.conf.SyncFlushing {
        log.Error("(*Agent).FlushSync called without apm_conf.sync_flushing enabled. No data was sent to Datadog.")
        return
    }
```
I'm wondering if we should crash completely because of this, as a critical error. It would seem like one. If you think so, simply change to `log.Critical`